integration-cli/daemon_swarm.go

```diff
@@ -33,6 +33,9 @@ var autoAcceptPolicy = swarm.AcceptancePolicy{
 func (d *SwarmDaemon) Init(req swarm.InitRequest) error {
 	if req.ListenAddr == "" {
 		req.ListenAddr = d.listenAddr
+		if strings.HasPrefix(req.ListenAddr, "0.0.0.0") {
```

Comment:
Currently: For join, swarmkit ends up choosing the address associated with the default route, which works in the unit test (just like before this PR).

In the near future: https://github.com/docker/swarmkit/pull/1121 will make address detection unnecessary for joining nodes (but it depends on this PR).

I originally had `Join` doing address detection too in this PR, but I removed it in anticipation of https://github.com/docker/swarmkit/pull/1121